xbmc/filesystem/VideoDatabaseDirectory/DirectoryNode.cpp

```diff
@@ -50,14 +50,14 @@ using namespace XFILE::VIDEODATABASEDIRECTORY;
 //  Constructor is protected use ParseURL()
 CDirectoryNode::CDirectoryNode(NODE_TYPE Type, const std::string& strName, CDirectoryNode* pParent)
 {
-  m_Type=Type;
-  m_strName=strName;
-  m_pParent=pParent;
+  m_Type = Type;
+  m_strName = strName;
+  m_pParent = pParent;
 }
 
 CDirectoryNode::~CDirectoryNode()
 {
-  delete m_pParent;
+  delete m_pParent, m_pParent = nullptr;
```

Comment:
please put the assignment on it's own line. You can actually overload operator , so this can blow up on us one day :)